### ISSUE TITLE
[Product] - Merge Flattened Product Schema

### ISSUE THREAD

**Issue by aaronjudd** [reactioncommerce/reaction#801]
A new data structure that doesn't embed variants but keeps the variant/product at the same level with a single definition (product).  This will increase performance and make it easier to deal with variants.

See https://github.com/reactioncommerce/reaction/pull/741 and https://github.com/reactioncommerce/reaction/issues/150

**Comment by aaronjudd** [comment]
@newsiberian please attach all the PR's for the new product catalog to this issue.

**Comment by newsiberian** [comment]
I've updated #741 with development. All tests passed. I suppose, it is ready to be merged.
Also, I moved `collection2-forks` to packages folder, so no need to fetch it manually from my repo.
`reaction-catalog`, also moved to `packages` folder.

So, all that needed to make it work now is in #741 PR + [this PR](https://github.com/reactioncommerce/reaction-factories/pull/5) contains stuff for tests. Also, we provide some docs which is [here](https://github.com/reactioncommerce/reaction-docs/pull/4)
